### DISCUSSION TITLE
lag when navigate to a new screen

### DISCUSSION THREAD

**Discussion by Pietro-Putelli** [react-navigation/react-navigation#10223] [General]
I'm using some screens that need to load a bit heavy data when they are presented, for example I've created a custom image picker that when navigate to it displays the first 30 images of user library, and that cause a lag during the presentation.
Another example is a ScrollView step screen, which has 6 child, and when presenting this screen the lag occurs again.

Since react native doesn't provide a multithreading, and the only library that allow it cause my app crashing, how can I solve this lag? Thank you

**Comment by ranaavneet** [comment]
Hi @Pietro-Putelli Did you find anything interesting to solve this problem ?

**Reply by Pietro-Putelli** [reply]
> Hi @Pietro-Putelli Did you find anything interesting to solve this problem ?

Yes, avoid this library, use react-native-navigation from wix, the drawback is that it use native modules so you can't use it with expo.
